sdk/purview/purview-sharing-rest/samples-dev/senderOperationsSample.ts

```diff
@@ -0,0 +1,171 @@
+          },
+          storeKind: storeKind,
+          storeReference: {
+            type: "ArmResourceReference",
+            referenceName: storageAccountResourceId,
+          },
+        },
+        displayName: `Sample-${storeKind}-Sent-Share`,
+      },
+      shareKind: "InPlace",
+    },
+  };
+
+  const initialResponse = await client.path("/sentShares/{sentShareId}", sentShareId).put(options);
+  const poller = await getLongRunningPoller(client, initialResponse);
+  const result = await poller.pollUntilDone();
+  console.log(result);
+
+  return result.body as InPlaceSentShareOutput;
+}
+
+/**
+ * This sample demonstrates how to create a recipient of Service type for a given sent share
+ *
+ * @summary Create a recipient for a given sent share
+ */
+async function createSentShareServiceInvitation(
+  client: PurviewSharingClient,
+  sentShareId: string,
+  sentShareInvitationId: string
+): Promise<SentShareInvitationOutput> {
+  const options: SentSharesCreateSentShareInvitationParameters = {
+    body: {
+      invitationKind: "Service",
+      properties: {
+        expirationDate: undefined,
+        targetActiveDirectoryId: "5DAE1226-9FAA-4D71-B8D4-87B81DFF672E",
+        targetObjectId: "EFA02830-BB7A-4586-B615-A6DFF19FEBBF",
+      },
+    },
+  };
+
+  const result = await client
+    .path(
+      "/sentShares/{sentShareId}/sentShareInvitations/{sentShareInvitationId}",
+      sentShareId,
+      sentShareInvitationId
+    )
+    .put(options);
+  console.log(result);
+
+  return result.body as SentShareInvitationOutput;
+}
+
+/**
+ * This sample demonstrates how to create a recipient of User type for a given sent share
+ *
+ * @summary Create a recipient for a given sent share
+ */
+async function createSentShareUserInvitation(
+  client: PurviewSharingClient,
+  sentShareId: string,
+  sentShareInvitationId: string
+): Promise<SentShareInvitationOutput> {
+  const options: SentSharesCreateSentShareInvitationParameters = {
+    body: {
+      invitationKind: "User",
+      properties: {
+        expirationDate: new Date("2025-07-21T23:52:00.7691109Z"),
+        notify: true,
+        targetEmail: "testReceiver@microsoft.com",
+      },
+    },
+  };
+
+  const result = await client
+    .path(
+      "/sentShares/{sentShareId}/sentShareInvitations/{sentShareInvitationId}",
+      sentShareId,
+      sentShareInvitationId
+    )
+    .put(options);
+  console.log(result);
+
+  return result.body as SentShareInvitationOutput;
+}
+
+async function main() {
+  const endpoint = process.env["ENDPOINT"] || "";
+  const blobStorageAccountResourceId = process.env["BLOB_STORAGE_ACCOUNT_RESOURCE_ID"] || "";
+  const adlsgen2StorageAccountResourceId =
+    process.env["ADLSGEN2_STORAGE_ACCOUNT_RESOURCE_ID"] || "";
+
+  const credential = new DefaultAzureCredential();
+  const client = createPurviewSharingClient(endpoint, credential);
+
+  // Create sent shares.
+  const sentShareForBlobAsset = await createOrReplaceSentShare(
```

Comment:
I'm wondering if we should just inline the content of these helper functions here? (i.e. `createOrReplaceSentShare`). What do you think?

Reply:
createOrReplaceSentShare is overloaded to be able to create two kinds of storage accounts supported (Blob and Gen2) and hence the helper. If they were to be inlined there would be duplicate code. Imo helpers make it more concise. But createSentShareXXXInvitation could be inlined. I thought of keeping all in helpers for consistency. Let me know your thoughts on this approach.